libraries/SocketWrapper/SocketHelpers.h

```diff
@@ -12,157 +12,51 @@
-            LOG_INF("    Router[%d]: %s", net_if_get_by_iface(iface),
-                net_addr_ntop(AF_INET,
-                            &iface->config.ip.ipv4->gw,
-                            buf, sizeof(buf)));
-            LOG_INF("Lease time[%d]: %u seconds", net_if_get_by_iface(iface),
-                iface->config.dhcpv4.lease_time);
-        }
-    }
-
-    static void option_handler(struct net_dhcpv4_option_callback *cb,
-                size_t length,
-                enum net_dhcpv4_msg_type msg_type,
-                struct net_if *iface)
-    {
-        char buf[NET_IPV4_ADDR_LEN];
-
-        LOG_INF("DHCP Option %d: %s", cb->option,
-            net_addr_ntop(AF_INET, cb->data, buf, sizeof(buf)));
-    }
-
-protected:
-
-    struct net_if *netif = nullptr;
-    int dhcp()
-    {
-        net_mgmt_init_event_callback(&mgmt_cb, event_handler, NET_EVENT_IPV4_ADDR_ADD | NET_EVENT_IF_UP | NET_EVENT_IF_DOWN);
-        net_mgmt_add_event_callback(&mgmt_cb);
-
-        net_dhcpv4_init_option_callback(&dhcp_cb, option_handler,
-                        DHCP_OPTION_NTP, ntp_server,
-                        sizeof(ntp_server));
-
-        net_dhcpv4_add_option_callback(&dhcp_cb);
-
-        net_dhcpv4_start(netif);
-
-        return 0;
-    }
-
-    void enable_dhcpv4_server(struct net_if *netif, char* _netmask = "255.255.255.0")
-    {
-        static struct in_addr addr;
-        static struct in_addr netmaskAddr;
-
-        if (net_addr_pton(AF_INET, String(localIP()).c_str(), &addr)) {
-            LOG_ERR("Invalid address: %s", String(localIP()).c_str());
-            return;
-        }
-
-        if (net_addr_pton(AF_INET,  _netmask, &netmaskAddr)) {
-            LOG_ERR("Invalid netmask: %s", _netmask);
-            return;
-        }
-
-        net_if_ipv4_set_gw(netif, &addr);
-
-        if (net_if_ipv4_addr_add(netif, &addr, NET_ADDR_MANUAL, 0) == NULL) {
-            LOG_ERR("unable to set IP address for AP interface");
-        }
-
-        if (!net_if_ipv4_set_netmask_by_addr(netif, &addr, &netmaskAddr)) {
-            LOG_ERR("Unable to set netmask for AP interface: %s", _netmask);
-        }
-
-        addr.s4_addr[3] += 10; /* Starting IPv4 address for DHCPv4 address pool. */
-
-        if (net_dhcpv4_server_start(netif, &addr) != 0) {
-            LOG_ERR("DHCP server is not started for desired IP");
-            return;
-        }
+    private:
+        uint8_t ntp_server[4];
+        static struct net_mgmt_event_callback mgmt_cb;
+        static struct net_dhcpv4_option_callback dhcp_cb;
 
-        LOG_INF("DHCPv4 server started...\n");
-    }
+        static void event_handler(struct net_mgmt_event_callback *cb,
+                    uint64_t mgmt_event,
+                    struct net_if *iface);
 
+        static void option_handler(struct net_dhcpv4_option_callback *cb,
+                    size_t length,
+                    enum net_dhcpv4_msg_type msg_type,
+                    struct net_if *iface);
```

Comment:
Could we make these methods into static functions in cpp file only? this way they are going to be hidden from the class definition